notebook/static/notebook/js/notebook.js

```diff
@@ -2876,10 +2876,13 @@ define([
                                 cell.output_area.trusted = true;
                             }
                         }
-                        nb.events.on('notebook_saved.Notebook', function () {
-                            window.location.reload();
+                        nb.contents.trust(nb.notebook_path)
+                            .then(function(res) {
+                                nb.events.trigger("trust_changed.Notebook", true);
+                                window.location.reload();
+                            }, function(err) {
+                                console.log(err);
                         });
-                        nb.save_notebook();
```

Comment:
So if its read-only, we save it and then trust..Thanks, never noticed the ``dirty`` attribute.